src/checkers/flashchecker.py

```diff
@@ -0,0 +1,130 @@
+    'Firefox': 'npapi',
+}
+
+
+class FlashChecker(Checker):
+    def __init__(self):
+        self._pkgs_cache = {}
+
+    def _should_check(self, external_data):
+        return external_data.checker_data.get('type') == 'flash'
+
+    def _get_arches(self, external_data):
+        if len(external_data.arches) != 1:
+            return None
+
+        flatpak_arch = external_data.arches[0]
+        return flatpak_arch, FLATPAK_TO_FLASH_ARCH_MAP.get(external_data.arches[0])
+
+    def check(self, external_data):
+        if not self._should_check(external_data):
+            log.debug('%s is not a flash type ext data', external_data.filename)
+            return
+
+        browser = external_data.checker_data['browser'].title()
+
+        try:
+            papi = BROWSER_TO_PAPI_MAP[browser]
+        except KeyError:
+            log.warning('%s has an invalid browser (should be Chrome or Firefox)',
+                        external_data.filename)
+            return
+
+        arches = self._get_arches(external_data)
+        if arches is None:
+            log.warning('%s has invalid only-arches (should be [i386] or [x86_64])',
+                        external_data.filename)
+            return
+
+        flatpak_arch, flash_arch = arches
+
+        request_params = {'platform_type': 'Linux', 'eventname': 'flashplayerotherversions',
+                          'platform_arch': flash_arch}
+        request_url = FLASH_BASE_URL.format(params=urllib.parse.urlencode(request_params))
+        with urllib.request.urlopen(request_url) as resp:
+            latest_version_data = json.load(resp)
+
+        latest_version = None
+        latest_url = None
+
+        for version in latest_version_data:
+            if (version['installation_type'] == 'Standalone' and version['browser'] == browser
+                    and version['installer_architecture'] == flash_arch):
+                assert version['platform'] == 'Linux'
+
+                latest_version = version['Version']
+                latest_url = version['download_url']
+                break
+
+        if latest_version is None:
+            log.warning(f'%s had no available URL', external_data.filename)
+            return
+        assert latest_url is not None
+
+        try:
+            _, data, checksum, size = utils.get_extra_data_info_from_url(latest_url)
+        except urllib.error.HTTPError as e:
+            log.warning('%s returned %s', url, e)
+            external_data.state = ExternalData.State.BROKEN
+        except Exception:
+            log.exception('Unexpected exception while checking %s', url)
+            external_data.state = ExternalData.State.BROKEN
+        else:
+            new_version = ExternalFile(latest_url, checksum, size, latest_version)
+
+            if external_data.current_version.matches(new_version):
+                external_data.state = ExternalData.State.VALID
+            else:
+                external_data.state = ExternalData.State.BROKEN
```

Comment:
I don't think it's quite right to set it as broken here – the old URL may still work. It's somewhat academic though

Reply:
Ah, I didn't realize it'll still show a new version if it's not marked as broken.

Reply:
(I think it will but I could be wrong!)